protocol.md

```diff
@@ -0,0 +1,233 @@
+# Mcumgr/Newtmgr protocol
+
+The `newtmgr` tool uses a custom protocol to send commands and responses
+between the manager and the end node using a variety of transports (currently
+TTY serial or BLE).
+
+The protocol isn't documented but the following information has been inferred
+from the source code available on Github and using the `-l DEBUG` flag When
+executing commands.
+
+## Source Code
+
+The golang source for **newtmgr** is [available here](https://github.com/apache/mynewt-newtmgr),
```

Comment:
So mcumgr, the tool, is the SMP client. mcumgr, the C library, is the SMP server

Reply:
Replace with a link to: https://github.com/apache/mynewt-mcumgr-cli